pyproject.toml

```diff
@@ -5,11 +5,11 @@ requires = [
     # order of magnitude slower, so avoid it for now.
     #
     "Cython>=0.29.32,<3.0.0",
-    # oldest supported Numpy for this platform is 1.17 but the oldest supported by Gensim
-    # is 1.18.5, remove the line when they increase oldest supported Numpy for this platform
-    "numpy==1.18.5; python_version=='3.8' and platform_machine not in 'arm64|aarch64'",
+    # oldest supported Numpy for non-'arm64|aarch64' architectures is 1.17.*
+    # as well as for 'Windows' on arm64. The oldest supported Numpy by Gensim
+    # is 1.18.5. Remove below line when they increase the oldest supported Numpy for those platforms
+    "numpy==1.18.5; python_version=='3.8' and platform_machine != 'aarch64' and (platform_machine != 'arm64' or platform_system == 'Windows')",
```

Comment:
What is the motivation behind changing this?

Reply:
Looking at the meta package [_oldest-supported-numpy_](https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg) I see that `numpy<1.18.5` will be downloaded for python 3.8 and Windows on arm64 (apart from all non-arm64 and non-aarch64 platforms). 

So I believe this check was incomplete resulting in installing unsupported version of Numpy for python3.8 arm64 Windows users.

Reply:
@piskvorky Do we want to deal with this now? It hasn't been reported yet, so it may not actually be a problem. It's been a while since I've messed with oldest-supported-numpy, so I'm not 100% sure this fix is needed and safe.

Reply:
I understand the reasoning. 
Should I revert this change then?

Reply:
Yes, let's revert it. Feel free to make a separate PR with that change so we can review it later, after the current release is out.

Reply:
I'll do as advised, thanks!